### PR TITLE
signing: allow disabling by setting backend to empty string

### PR DESCRIPTION
**Pull request by martinvonz**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Comment by martinvonz** [comment]
I should make `signing.backend` also default to `""` and document that. I'll make that change

**Comment by martinvonz** [comment]
> I should make `signing.backend` also default to `""` and document that. I'll make that change

Done

File: lib/src/settings.rs [hunk 1]
```diff
@@ -267,7 +267,11 @@ impl UserSettings {
     // separate from sign_settings as those two are needed in pretty different
     // places
     pub fn signing_backend(&self) -> Option<String> {
-        self.config.get_string("signing.backend").ok()
+        let backend = self
+            .config
+            .get_string("signing.backend")
```
Comment (ilyagr) [inline lib/src/settings.rs:272]: I wonder whether

   .get_string(...)?;

would be clearer than `unwrap_or_default` (assuming it works, of course).

**Review by ilyagr — APPROVED** [review]

File: lib/src/settings.rs (hunk 1 above)
Comment (martinvonz) [reply lib/src/settings.rs]: I'm not sure it's clearer but it's at least shorter (and not less clear). Good idea.

File: docs/config.md [hunk 2]
```diff
@@ -776,6 +776,8 @@ GnuPG or SSH signing keys.
 
 To do this you need to configure a signing backend.
 
+Setting the backend to `""` (empty string) disables signing.
```
Comment (yuja) [inline docs/config.md:779]: nit: might be better to use `"none"`. `signing.backend` is enum-like, and is similar to `core.fsmonitor`.

File: lib/src/settings.rs (hunk 1 above)
Comment (ilyagr) [reply lib/src/settings.rs]: Another option is `.get_string(...).ok().filter(|x| x != "none")`.

I'll unresolve the conversation, but feel free to use it or not.

Comment (martinvonz) [reply lib/src/settings.rs]: I went with `.ok()?` and a subsequent `.then_some()`

File: docs/config.md (hunk 2 above)
Comment (martinvonz) [reply docs/config.md]: Makes sense. Done.

**Pull request merged**